### ISSUE TITLE
Count must be non-negative when purging durable metadata

### ISSUE THREAD

**Issue by jhueppauff** [microsoft/durabletask-dotnet#111]
Hi, 

When I try to clean up the durable metadata but no instance matches the filter I get the following exception.
`await client.Client.PurgeInstancesAsync(filter: new PurgeInstancesFilter(createdTimeFrom, createdTimeTo, states));`

```
[2023-01-25T15:39:01.222Z] Executed 'Functions.CleanupOldWorkflows' (Failed, Id=ff84dea6-2dd1-4cf0-8e23-a3d6ba3bc491, Duration=5787ms)
[2023-01-25T15:39:01.223Z] System.Private.CoreLib: Exception while executing function: Functions.CleanupOldWorkflows. System.Private.CoreLib: Result: Failure
Exception: System.AggregateException: One or more errors occurred. (Count must be non-negative (Parameter 'count'))
[2023-01-25T15:39:01.224Z]  ---> System.ArgumentException: Count must be non-negative (Parameter 'count')
[2023-01-25T15:39:01.225Z]    at Microsoft.DurableTask.Check.Argument(Boolean condition, String name, String message, Object[] args)
[2023-01-25T15:39:01.226Z]    at Microsoft.DurableTask.Client.PurgeResult..ctor(Int32 count)
[2023-01-25T15:39:01.226Z]    at Microsoft.DurableTask.Client.Grpc.GrpcDurableTaskClient.PurgeInstancesCoreAsync(PurgeInstancesRequest request, CancellationToken cancellation)
```

It seems to be related to this line:
https://github.com/microsoft/durabletask-dotnet/blob/main/src/Client/Core/PurgeResult.cs#L17

Shouldn't the condition be `count >= 0`? This way an exception is thrown when no instance was removed.

I can also file a PR if needed for this :)

**Comment by jviau** [comment]
Thank you for finding this! You are correct it should be `count >= 0`. Unfortunately, we had already prepared the `1.0.0` release so this will need to go into the next release. If you want to make a PR for it go ahead.

**Comment by jhueppauff** [comment]
Done :) #115

**Comment by jviau** [comment]
Resolved by #115
